### PR TITLE
Update docs. Fixed installation instructions.

### PR DESCRIPTION
**Pull request by JSKenyon**

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -2,61 +2,63 @@
...
-	cloned the repository, this can be done as follows:
+.. note::
 
-	.. code:: bash
+	CubiCal predicts model visiblities using Montblanc_. The CPU version of Montblanc is
+	installed automatically. To install the GPU version of Montblanc, follow the
+	instructions here_ before installing cubical.
+
+	.. _Montblanc: https://arxiv.org/abs/1501.07719
+	.. _here: https://montblanc.readthedocs.io
```
Comment (bennahugo) [inline docs/installation.rst]: These instructions are out of date. Use the setup.py optional dependencies

File: docs/installation.rst [hunk 2]
```diff
@@ -2,61 +2,63 @@
...
 
-CubiCal can now be installed from PyPI by running the following:
+Once the ppa has been added, CubiCal's dependencies can be installed as
+follows:
 
 .. code:: bash
 
-	pip install cubical
+	CUBICAL_DEPENDENCIES=(casacore-dev \
+                     	      casacore-data \
```
Comment (bennahugo) [inline docs/installation.rst]: This version of casacore has a known deficiency and only reads chunks up to about 8 gb of data without corruption. I would recommend changing this to build and install casacore from source

File: docs/parset.rst [hunk 3]
```diff
@@ -185,340 +227,405 @@ This computes the median absolute residual (i.e. median absolute deviation from
...
-	Generate summary plots. Default: 1.
+    Which model directions to subtract, if generating
+    residuals. ":" subtracts all. Can also be specified as
+    "N", "N:M", ":N", "N:", "N,M,K". Default: :.
+--out-plots=bool
+    Generate summary plots. Default: 1.
 --out-plots-show=bool
-	Show summary plots interactively. Default: 0.
+    Show summary plots interactively. Default: 0.
 --out-casa-gaintables=bool
```
Comment (bennahugo) [inline docs/parset.rst]: we should remove this option. Python casacore 3 no longer supports writing the necessary metadata to these tables.

File: docs/installation.rst (hunk 1 above)
Comment (bennahugo) [reply docs/installation.rst]: specifically cubical does not work with the master branch

File: docs/parset.rst (hunk 3 above)
Comment (JSKenyon) [reply docs/parset.rst]: I will relegate this change to a future PR - currently we print warnings everywhere anyway. This PR only fixes the docs to be more consistent with the -h option.

File: docs/installation.rst (hunk 2 above)
Comment (JSKenyon) [reply docs/installation.rst]: I have left the simple instructions intact for now but added a warning callout box which instructs the user to install from source if they are doing large reads.

File: docs/installation.rst (hunk 1 above)
Comment (JSKenyon) [reply docs/installation.rst]: I have updated the installation instructions to include the optional Montblanc dependency.

**Comment by JSKenyon** [comment]
@bennahugo I believe I have addressed you comments.

**Pull request merged**